Sources/StructuralCore/Structural.swift

```diff
@@ -25,57 +27,55 @@ public protocol Structural {
 }
 
 /// Structural representation of a Swift struct.
-public struct StructuralStruct<Properties> {
-    public var type: Any.Type?
+public struct StructuralStruct<BaseType, Properties> {
+    public var type: BaseType.Type?
     public var properties: Properties
 
     public init(_ properties: Properties) {
         self.type = nil
         self.properties = properties
     }
 
-    public init(_ type: Any.Type, _ properties: Properties) {
+    public init(_ type: BaseType.Type, _ properties: Properties) {
         self.type = type
         self.properties = properties
     }
 }
 
 /// Structural representation of a Swift property.
-public struct StructuralProperty<Value> {
+public struct StructuralProperty<BaseType, Value, WrappedValue> {
```

Comment:
Please document `Value` vs `WrappedValue`. (I only understood the distinction when looking through the examples and seeing one using property wrappers.)

Agreed that this is some unfortunate complexity. :-(

Reply:
Added comment in b20fb72